### PR TITLE
[docs] AutoPipeline

### PR DESCRIPTION
**Pull request by stevhliu**
Refactors the `from_pipe` API from the AutoPipeline tutorial to link to the [Reuse a pipeline](https://huggingface.co/docs/diffusers/main/en/using-diffusers/loading#reuse-a-pipeline) section and cleans up the AutoPipeline API doc.

### CODE REVIEW

File: docs/source/en/api/pipelines/auto_pipeline.md [hunk 1]
```diff
@@ -12,42 +12,10 @@ specific language governing permissions and limitations under the License.
...
-
-`AutoPipeline` supports text-to-image, image-to-image, and inpainting for the following diffusion models:
-
-- [Stable Diffusion](./stable_diffusion/overview)
-- [ControlNet](./controlnet)
-- [Stable Diffusion XL (SDXL)](./stable_diffusion/stable_diffusion_xl)
-- [DeepFloyd IF](./deepfloyd_if)
-- [Kandinsky 2.1](./kandinsky)
-- [Kandinsky 2.2](./kandinsky_v22)
+The `AutoPipeline` is designed to make it easy to load a checkpoint for a task without needing to know the specific pipeline class. Based on the task, the `AutoPipeline` automatically retrieves the correct pipeline class from the checkpoint model_index.json file.
```
Comment (sayakpaul) [inline docs/source/en/api/pipelines/auto_pipeline.md:15]:
```suggestion
The `AutoPipeline` is designed to make it easy to load a checkpoint for a task without needing to know the specific pipeline class. Based on the task, the `AutoPipeline` automatically retrieves the correct pipeline class from the checkpoint `model_index.json` file.
```

File: docs/source/en/tutorials/autopipeline.md [hunk 2]
```diff
@@ -12,75 +12,69 @@ specific language governing permissions and limitations under the License.
...
-    <img src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/diffusers/autopipeline-text2img.png" alt="generated image of peasant fighting dragon in wood cutting style"/>
+    <img src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/diffusers/autopipeline-text2img.png"/>
 </div>
 
-Under the hood, [`AutoPipelineForText2Image`]:
-
-1. automatically detects a `"stable-diffusion"` class from the [`model_index.json`](https://huggingface.co/runwayml/stable-diffusion-v1-5/blob/main/model_index.json) file
-2. loads the corresponding text-to-image [`StableDiffusionPipeline`] based on the `"stable-diffusion"` class name
-
-Likewise, for image-to-image, [`AutoPipelineForImage2Image`] detects a `"stable-diffusion"` checkpoint from the `model_index.json` file and it'll load the corresponding [`StableDiffusionImg2ImgPipeline`] behind the scenes. You can also pass any additional arguments specific to the pipeline class such as `strength`, which determines the amount of noise or variation added to an input image:
```
Comment (sayakpaul) [inline docs/source/en/tutorials/autopipeline.md:53]: Why is this going away?

File: docs/source/en/tutorials/autopipeline.md [hunk 3]
```diff
@@ -12,75 +12,69 @@ specific language governing permissions and limitations under the License.
...
+generator = torch.Generator(device="cpu").manual_seed(53)
+image = pipeline(prompt, image=init_image, generator=generator).images[0]
+image
+```
 
-image = pipeline(prompt, image, num_inference_steps=200, strength=0.75, guidance_scale=10.5).images[0]
+Notice how the [dreamlike-art/dreamlike-photoreal-2.0](https://hf.co/dreamlike-art/dreamlike-photoreal-2.0) checkpoint is used for both text-to-image and image-to-image tasks? To save memory and avoid loading the checkpoint twice, use the [`~DiffusionPipeline.from_pipe`] method.
+
+```py
+pipe_img2img = AutoPipelineForImage2Image.from_pipe(pipeline).to("cuda")
```
Comment (sayakpaul) [inline docs/source/en/tutorials/autopipeline.md:65]: `pipeline` already an img2img pipeline no? 

```
pipeline = AutoPipelineForImage2Image.from_pretrained(
    "dreamlike-art/dreamlike-photoreal-2.0", torch_dtype=torch.float16, use_safetensors=True
).to("cuda")
```

**Review by sayakpaul — COMMENTED** [review]
Thank you. Left some comments.

**Review by yiyixuxu — APPROVED** [review]
thanks!

File: docs/source/en/tutorials/autopipeline.md (hunk 2 above)
Comment (stevhliu) [reply docs/source/en/tutorials/autopipeline.md:53]: This is repackaged and explained in the third paragraph so it's still there :)

File: docs/source/en/tutorials/autopipeline.md (hunk 3 above)
Comment (stevhliu) [reply docs/source/en/tutorials/autopipeline.md:65]: Should be clearer and fixed now. The `pipeline` referenced here was from the text-to-image example.

**Review by sayakpaul — APPROVED** [review]
Gracias

**Pull request merged**